addon/components/behave/scrollable.hbs

```diff
@@ -0,0 +1,30 @@
+<div
+  class="
+    Behave-Scrollable
+    -{{this.direction}}
+    {{if this.isEnabled "-enabled"}}
+    {{if this.isAtStart "-start"}}
+    {{if this.isAtEnd "-end"}}
+  "
+
+  ...attributes
+>
+  <div class="Behave-Scrollable-Start {{this.startClass}}">
+    {{yield to="start"}}
+  </div>
+
+  <div class="Behave-Scrollable-End {{this.endClass}}">
+    {{yield to="end"}}
+  </div>
```

Comment:
> Why do we need those two divs and yields? I think it is making assumptions here about the usage, and also is less "headless" than possible. And we are not applying any special behavior to these, right?

I went through a number of iterations while developing this feature. This is kinda a remnant from a previous effort.

Generally though, here's how I see conditional components VS named yields:

<table>
  <tr>
    <th></th>
    <th>Contextual components</th>
    <th>Named yields</th>
  </tr>

  <tr>
    <th>Applying HTML classes</th>
    <td>💚</td>
    <td>💔</td>
  </tr>

  <tr>
    <th>Overhead</th>
    <td>💔</td>
    <td>💚</td>
  </tr>

  <tr>
    <th>Can be omitted when only the default block is neeed</th>
    <td>💔</td>
    <td>💚</td>
  </tr>

  <tr>
    <th>Positioning within parent's markup</th>
    <td>💔</td>
    <td>💚</td>
  </tr>
</table>

The last point is what makes me the most concerned: certain features are simply impossible without named yields, without forcing the user to reproduce a certain layout within the yielded block. We're just lucky that this component has relatively simple HTML.

***

> Also we don't currently have a way to render things conditionally, as we don't yield the is* states.

That's because using `{{#unlees}}` will remove fade animation. But yielding the booleans won't hurt, I guess.

***

> I believe this is necessary for the same reason that you probably also dealt with, that the indicators must not be part of the scrollable area itself, right?

Yes, without an extra wrapper, indicators would scroll away.

Also, lack of a wrapper would break `:first-child` and `:last-child` pseudoselectors for content.

***

I'll adjust the component to your liking.

Reply:
Done.

PS Another rant toward yielded components is that typos in their invocations fail silently, whereas invalid yield targets produce eager errors.